### PR TITLE
Add addCustomAttribute to EC Class and Property method to editing API

### PR DESCRIPTION
**Pull request by christophermlawson**

### CODE REVIEW

File: core/ecschema-editing/src/test/Editing/ECClasses.test.ts [hunk 1]
```diff
@@ -362,4 +362,164 @@ describe("Property creation tests", () => {
     property = await entity?.getProperty(propResult.propertyName!) as PrimitiveProperty;
     expect(property).to.be.undefined;
   });
+
+  it.only("CustomAttribute defined in same schema, instance added to class successfully.", async () => {
```
Comment (pmconne) [inline core/ecschema-editing/src/test/Editing/ECClasses.test.ts:366]: remove `.only` here and elsewhere.

**Comment by pmconne** [comment]
Have you considered replacing this:
```
export interface SchemaItemEditResults {
  itemKey?: SchemaItemKey;
  errorMessage?: string;
}
```

With this:
```
export interface SchemaItemEditSuccess {
  itemKey: SchemaItemKey;
  errorMessage?: never;
}

export interface SchemaItemEditError {
  errorMessage: string;
  itemKey?: never;
}

export type SchemaItemEditResults = SchemaItemEditSuccess | SchemaItemEditError;
```

To accurately reflect that the result is exactly one of an item key or an error message, never both nor neither?
This also makes calling code much simpler because the compiler knows that `result.errorMessage` will be `undefined ` IFF `result.itemKey` is `undefined`.

File: core/ecschema-editing/src/test/Editing/ECClasses.test.ts (hunk 1 above)
Comment (christophermlawson) [reply core/ecschema-editing/src/test/Editing/ECClasses.test.ts]: Done. Thanks.

**Comment by christophermlawson** [comment]
@pmconne Thanks for the suggestion. I've created a backlog item and will refactor the code soon.

File: core/ecschema-editing/src/test/Editing/ECClasses.test.ts [hunk 2]
```diff
@@ -362,4 +362,164 @@ describe("Property creation tests", () => {
     property = await entity?.getProperty(propResult.propertyName!) as PrimitiveProperty;
     expect(property).to.be.undefined;
   });
+
+  it("CustomAttribute defined in same schema, instance added to class successfully.", async () => {
```
Comment (ColinKerr) [inline core/ecschema-editing/src/test/Editing/ECClasses.test.ts:366]: Add tests where you try to add a custom attribute to a class or property that does not exist.  Also add a test where you try to add a custom attribute to a schema item that does not support custom attributes (like KOQ)

Comment (christophermlawson) [reply core/ecschema-editing/src/test/Editing/ECClasses.test.ts:366]: Sorry, need one more test....

**Review by pmconne — APPROVED** [review]

File: core/ecschema-editing/src/test/Editing/ECClasses.test.ts (hunk 2 above)
Comment (christophermlawson) [reply core/ecschema-editing/src/test/Editing/ECClasses.test.ts:366]: New tests added.

**Review by ColinKerr — APPROVED** [review]

**Pull request merged**